src/lae_site/handlers/create_subscription.py

```diff
@@ -111,46 +120,50 @@ def create_customer(self, stripe_authorization_token, user_email):
                     "engineers have been notified and will contact you with "
-                    "an update shortly.",
+                    "an update shortly."
                 ),
                 email_subject="Stripe Invalid Request error ({})".format(user_email),
             )
+        except stripe.AuthenticationError as e:
+            request.setResponseCode(UNAUTHORIZED)
+            self.handle_stripe_create_customer_errors(
+                traceback.format_exc(100), e,
+                details=(
+                    "Our payment processor is temporarily unavailable, "
+                    "please try again in a few moments."
+                ),
+                email_subject="Stripe Auth error ({})".format(user_email),
+            )
         except Exception as e:
+            # Return a generic error here
+            request.setResponseCode(BAD_REQUEST)
             self.handle_stripe_create_customer_errors(
                 traceback.format_exc(100), e,
                 details=(
                     "Something went wrong. Please try again, or contact "
-                    " <support@leastauthority.com>."
+                    "support@leastauthority.com."
                 ),
                 email_subject="Stripe unexpected error ({})".format(user_email),
             )
 
+    def render_POST(self, request):
+        request = self.addHeaders(request, self._cross_domain)
+        stripe_authorization_token = request.args.get(b"stripeToken")[0]
+        user_email = request.args.get(b"email")[0]
 
-    def render(self, request):
-        # The expected HTTP method is a POST from the <form> in templates/subscription_signup.html.
-        # render_POST is handled by the HandlerBase parent which calls this this method after logging
-        # the request.
-
-        if request.method != 'POST':
-            tmpl = env.get_template('s4-subscription-form.html')
-            msg = ("Your browser requested this page with {} but we expected a POST."
-                   "\n\nPlease try again.".format(request.method))
-            return tmpl.render({"errorblock": msg}).encode('utf-8', 'replace')
-
-        # Get information needed to create the new stripe subscription to the S4 plan
-        stripe_authorization_token, user_email = self.get_creation_parameters(request)
         try:
             # Invoke card charge by requesting subscription to recurring-payment plan.
-            customer = self.create_customer(stripe_authorization_token, user_email)
+            customer = self.create_customer(stripe_authorization_token, user_email, request)
         except RenderErrorDetailsForBrowser as e:
-            tmpl = env.get_template('s4-subscription-form.html')
-            return tmpl.render({"errorblock": e.details}).encode('utf-8', 'replace')
+            return e.details
 
         # Initiate the provisioning service
         subscription = customer.subscriptions.data[0]
```

Comment:
As far as I can tell, the cookie-based signup-style selection logic that starts on the next line no longer works.  Signup is always via email even if the cookie is set to wormhole.  Do you have any idea about why this might be?  Something related to CORS?